src/octopeer-github/main/Options/Options.ts

```diff
@@ -120,102 +92,31 @@ const Options = new (class Options {
     }
 
     /**
-     * Gets the logging preference, from the chrome storage.
-     * @returns {Boolean}
-     */
-    public getLogging() {
-        return this.logging;
-    }
-
-    /**
-     * Gets the tab tracking preference, from the chrome storage.
-     * @returns {Boolean}
-     */
-    public getTabs() {
-        return this.tabs;
-    }
-
-    /**
-     * Gets the comment tracking preference, from the chrome storage.
-     * @returns {Boolean}
-     */
-    public getComments() {
-        return this.comments;
-    }
-
-    /**
-     * Gets the peer comments tracking preference, from the chrome storage.
-     * @returns {Boolean}
-     */
-    public getPeerComments() {
-        return this.peerComments;
-    }
-
-    /**
-     * Gets the focus tracking preference, from the chrome storage.
-     * @returns {Boolean}
-     */
-    public getFocus() {
-        return this.focus;
-    }
-
-    /**
-     * Gets the username encryption preference, from the chrome storage.
-     * @returns {Boolean}
+     * Generates an array containing all option names, based on the optionMap.
+     * @returns {Array<String>} the list of option names.
      */
-    public getUsername() {
-        return this.username;
-    }
-
-    /**
-     * Gets the repository encryption preference, from the chrome storage.
-     * @returns {Boolean}
-     */
-    public getRepo() {
-        return this.repo;
-    }
-
-    /**
-     * Gets the file encryption preference, from the chrome storage.
-     * @returns {Boolean}
-     */
-    public getFile() {
-        return this.file;
-    }
-
-    /**
-     * Gets doNotWatch preference about onscreen elements, from the chrome storage.
-     * User perspective: Do not watch what elements are on my screen.
-     * @returns {Boolean}
-     */
-    public getDoNotWatchOnScreenEvents() {
-        return this.doNotWatchOnScreenEvents;
-    }
-
-    /**
-     * Gets doNotWatch preference about hovering above elements, from the chrome storage.
-     * User perspective: Do not watch what elements I hover over.
-     * @returns {Boolean}
-     */
-    public getDoNotWatchHoverEvents() {
-        return this.doNotWatchHoverEvents;
-    }
-
-    /**
-     * Gets doNotWatch preference about comments, from the chrome storage.
-     * User perspective: Do not watch the comments of my pull request.
-     * @returns {Boolean}
-     */
-    public getDoNotWatchCommentElements() {
-        return this.doNotWatchCommentElements;
+    public generateOptionList(): string[] {
```

Comment:
generateOptionList is used for creating an object containing all chrome storage names.
Just to be sure that this is clear :)